src/AppInstallerCommonCore/AppInstallerTelemetry.cpp

```diff
@@ -91,11 +84,50 @@ namespace AppInstaller::Logging
         m_isRuntimeEnabled = true;
     }
 
+    void TelemetryTraceLogger::SetUserSettingsStatus()
+    {
+        m_isSettingEnabled = !Settings::User().Get<Settings::Setting::TelemetryDisable>();
+    }
+
+    void TelemetryTraceLogger::SetTelemetryCorelationJson(std::string jsonStr)
+    {
+        m_telemetryCorelationJson = jsonStr;
+    }
+
+    std::string TelemetryTraceLogger::GetTelemetryCorelationJson() const
```

Comment:
True. That was my first thought too. But I wanted to log the passed in Json even if invalid to FileLogger for debugging purpose. And went with this solution since I am only querying for it once per context in LogStartUp only.

But since we want to instead log the Json in every event, that will make it inefficient to have that check for invalid Json for every event logged. So, I will change move the check to Set method and log the invalid Json to FileLogger in Set itself.